apps/dashboard/src/utils/self-hosted/user.resource.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { createContextHook } from '../context';
+import { DecodedJwt } from '.';
+
+export const UserContext = React.createContext({});
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function UserContextProvider({ children }: any) {
+  const jwt = localStorage.getItem('self-hosted-jwt');
+  const decodedJwt: DecodedJwt | null = jwt ? JSON.parse(atob(jwt.split('.')[1])) : null;
+  const value = {
+    user: {
+      update: async () => null,
+      externalId: decodedJwt?._id,
+      firstName: decodedJwt?.firstName,
+      lastName: decodedJwt?.lastName,
+      emailAddresses: [{ emailAddress: decodedJwt?.email }],
+      createdAt: new Date(),
+      publicMetadata: { newDashboardOptInStatus: 'opted_in' },
+      unsafeMetadata: { newDashboardOptInStatus: 'opted_in' },
+      organizationMemberships: [{}],
+      passwordEnabled: true,
+    },
```

Comment:
Saw a duplication of this object somewhere else, can we combine them into a single mapped function?